core-rust/state-manager/src/state_manager.rs

```diff
@@ -235,13 +236,18 @@ where
             preview_intent,
         )
     }
+}
 
+impl<S> StateManager<S>
+where
+    S: ReadableSubstateStore + ReadableTreeStore,
+{
     fn execute_with_cache(
```

Comment:
I'd possibly support calling this `execute_for_staging_with_cache` or something - to distinguish from the mempool execution